src/components/transport_manager/src/iap2_emulation/iap2_transport_adapter.cc

```diff
@@ -58,17 +73,86 @@ IAP2USBEmulationTransportAdapter::IAP2USBEmulationTransportAdapter(
     const uint16_t port,
     resumption::LastState& last_state,
     const TransportManagerSettings& settings)
-    : TcpTransportAdapter(port, last_state, settings) {}
+    : TcpTransportAdapter(port, last_state, settings), out_(0) {
+  auto delegate = new IAPSignalHandlerDelegate(this);
+  signal_handler_ =
+      threads::CreateThread("iAP signal handler", delegate);
+  signal_handler_->start();
+  LOG4CXX_DEBUG(logger_, "Out signals channel creation result: "
+                << mkfifo(out_signals_channel, mode));
+}
+
+IAP2USBEmulationTransportAdapter::~IAP2USBEmulationTransportAdapter() {
+  signal_handler_->join();
+  threads::DeleteThread(signal_handler_);  
+  LOG4CXX_DEBUG(logger_, "Out close result: " << close(out_));
+  LOG4CXX_DEBUG(logger_, "Out unlink result: " << unlink(out_signals_channel));
+}
 
 void IAP2USBEmulationTransportAdapter::DeviceSwitched(
     const DeviceUID& device_handle) {
   LOG4CXX_AUTO_TRACE(logger_);
   UNUSED(device_handle);
+  const auto switch_signal_ack = std::string("SDL_TRANSPORT_SWITCH_ACK\n");
```

Comment:
@dev-gh in that case explicit typing is better than explicit constructor call : 
```
const std::string switch_signal_ack("SDL_TRANSPORT_SWITCH_ACK");
```
Also, do really you need `\n`?

Reply:
@LuxoftAKutsan 
Yes because closing of pipe happens only on adapter instance destruction and thus reading the pipe never reached EOF so blocked. In case we have EOL Lua script can get this data, unblock and continue.